plc4j/spi/src/main/java/org/apache/plc4x/java/spi/connection/SingleProtocolStackConfigurer.java

```diff
@@ -142,31 +141,86 @@ public SingleProtocolStackBuilder(Class<BASE_PACKET_CLASS> basePacketClass, Clas
         }
 
         public SingleProtocolStackBuilder<BASE_PACKET_CLASS> withProtocol(Class<? extends Plc4xProtocolBase<BASE_PACKET_CLASS>> protocol) {
-            this.protocol = protocol;
+            return withProtocol(configuredType(protocol));
```

Comment:
I have to admit, that I would prefer if we refactored the existing drivers to use the new version instead of doing this.

Reply:
I think we can safely follow @sruehl suggestion to have two versions of stack configurers (or at least builders) - one which supports injections and other which does not.